src/huggingface_hub/repocard.py

```diff
@@ -0,0 +1,38 @@
+import re
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+import yaml
+
+
+REGEX_YAML_BLOCK = re.compile(r"---[\n\r]+([\S\s]*?)[\n\r]+---[\n\r]")
+# exact same regex as in the hf hub server. Please keep in sync.
```

Comment:
nit:

```suggestion
# exact same regex as in the Hub server. Please keep in sync.
REGEX_YAML_BLOCK = re.compile(r"---[\n\r]+([\S\s]*?)[\n\r]+---[\n\r]")
```

Also it would be good to add similar docstring in the Hub code